tests/jballerina-unit-test/src/test/resources/test-src/bala/test_projects/test_project_records/closed_record_type_reference.bal

```diff
@@ -39,3 +39,8 @@ public type ClosedVehicleWithNever record {|
     int j;
     never p?;
 |};
+
```

Comment:
Please add more tests
1. with multiple levels of inclusion: e.g.,
    i. baz:C includes bar:B which includes foo:A, where the default value is specified in foo
    ii. baz:C includes bar:B which includes foo:A, where the default value is specified in bar
2. to make sure the default value is used only when a value is not specified
3. a record includes two records with the same field (where one or both have a default), but the including record has to specify the field explicitly then. In such cases, add tests for both when the including record's field has a default value and doesn't have a default value. In either case default values from the included records shouldn't be used.

Reply:
added with https://github.com/ballerina-platform/ballerina-lang/pull/42428/commits/4d48e7ae41d789fe34cef3db98af0823ad10a889

Reply:
We aren't testing 2 (in conjunction with 3 also) are we? I guess that's OK since we have tests for the normal case.

Please add a test like 1, where foo:A has a default for a field which is overridden by bar:B, also with a default.

Reply:
Added with https://github.com/ballerina-platform/ballerina-lang/pull/42428/commits/6094bb847eee5c8a768c81e7b5fb60a8706b1fc7